### PR TITLE
Feature/delete item functionality

### LINKED ISSUE #27: Add ability to remove pantry item

### LINKED ISSUE #54: Implement delete functionality and related gesture
Gesture could be a "long press" or "swipe to left"

### PR DESCRIPTION
**Pull request by henrijaeger**
Closes #54
Closes #27

- [x] implement 'deleteItem' function in ItemController
- [x] add support for swipe gesture in PantryFragment

grafik
grafik

### CODE REVIEW

**Review by frcroth — APPROVED** [review]
Looks like it works, but everything is black now, why?

**Comment by frcroth** [comment]
Just like in your screenshot, and previously my background was white

**Comment by frcroth** [comment]
It is tied to dark mode and if I set it to light mode (which it never was
before) it looks like it used too do. Weird, but I guess it's fine

File: app/src/main/java/app/wefridge/wefridge/model/ItemController.kt [hunk 1]
```diff
@@ -44,12 +44,24 @@ class ItemController {
                 }
         }
 
-        fun deleteItem(item: Item) {
-            TODO("Not yet implemented")
+        fun deleteItem(item: Item, callbackOnSuccess: () -> kotlin.Unit, callbackOnFailure: (Exception) -> kotlin.Unit) {
+            callbackOnFailure(Exception())
```
Comment (PassiDel) [inline app/src/main/java/app/wefridge/wefridge/model/ItemController.kt:48]:
```suggestion
```

File: app/src/main/java/app/wefridge/wefridge/model/ItemController.kt [hunk 2]
```diff
@@ -44,12 +44,24 @@ class ItemController {
...
+                if (item.firebaseId == null)
+                    return@getUser
+
+                itemsRef
+                    .document(item.firebaseId!!)
+                    .delete()
+                    .addOnSuccessListener { callbackOnSuccess() }
+                    .addOnFailureListener { exception -> callbackOnFailure(exception) }
+
+            }, {})
```
Comment (PassiDel) [inline app/src/main/java/app/wefridge/wefridge/model/ItemController.kt:61]: Result is ignored.
```suggestion
            val itemsRef = FirebaseFirestore.getInstance().collection(ITEMS_COLLECTION_NAME)

            if (item.firebaseId == null)
                return callbackOnFailure(Exception())

            itemsRef
                .document(item.firebaseId!!)
                .delete()
                .addOnSuccessListener { callbackOnSuccess() }
                .addOnFailureListener { exception -> callbackOnFailure(exception) }
```

**Review by PassiDel — APPROVED** [review]

**Pull request merged**